MekHQ/src/mekhq/service/AutosaveService.java

```diff
@@ -98,39 +119,51 @@ private String getAutosaveFilename(Campaign campaign) {
         String savesDirectoryPath = MekHQ.getCampaignsDirectory().getValue();
         File folder = new File(savesDirectoryPath);
-        List<File> autosaveFiles = Arrays.stream(folder.listFiles())
-                .filter(f -> f.getName().startsWith("Autosave-"))
-                .sorted(Comparator.comparing(f -> f.lastModified()))
-                .collect(Collectors.toList());
-
-        // Delete older autosave files if needed
-        int maxNumberAutosaves = this.userPreferences.getInt(MekHqConstants.MAXIMUM_NUMBER_SAVES_KEY, MekHqConstants.DEFAULT_NUMBER_SAVES);
-        while (autosaveFiles.size() >= maxNumberAutosaves && autosaveFiles.size() > 0) {
-            autosaveFiles.get(0).delete();
-            autosaveFiles.remove(0);
-        }
+        File[] files = folder.listFiles();
+        if (files != null) {
+            List<File> autosaveFiles = Arrays.stream(files)
+                    .filter(f -> f.getName().startsWith("Autosave-"))
+                    .sorted(Comparator.comparing(File::lastModified))
+                    .collect(Collectors.toList());
+
+            // Delete older autosave files if needed
+            int maxNumberAutosaves = this.userPreferences.getInt(MekHqConstants.MAXIMUM_NUMBER_SAVES_KEY,
+                    MekHqConstants.DEFAULT_NUMBER_SAVES);
+
+            int index = 0;
+            while (autosaveFiles.size() >= maxNumberAutosaves && autosaveFiles.size() > index) {
+                if (autosaveFiles.get(index).delete()) {
+                    autosaveFiles.remove(index);
+                } else {
+                    this.logger.error(this.getClass(), "getAutosaveFilename",
+                            "Unable to delete file " + autosaveFiles.get(index).getName());
+                    index++;
+                }
+            }
 
-        // Find a unique name for this autosave
-        String fileName = null;
-
-        boolean repeatedName = true;
-        int index = 0;
-        while (repeatedName) {
-            fileName = String.format(
-                    "Autosave-%d-%s-%s.cpnx.gz",
-                    index++,
-                    campaign.getName(),
-                    campaign.getShortDateAsString());
-
-            repeatedName = false;
-            for (File file : autosaveFiles) {
-                if (file.getName().compareToIgnoreCase(fileName) == 0) {
-                    repeatedName = true;
-                    break;
+            // Find a unique name for this autosave
+            String fileName = null;
+
+            boolean repeatedName = true;
+            index = 1;
+            while (repeatedName) {
+                fileName = String.format(
+                        "Autosave-%d-%s-%s.cpnx.gz",
+                        index++,
+                        campaign.getName(),
+                        campaign.getShortDateAsString());
+
+                repeatedName = false;
+                for (File file : autosaveFiles) {
```

Comment:
Seems like if you turned autosaveFiles into a map, you could get an O(1) run time here, rather than O(#files)

Reply:
Given an expected n of 5, I would expect it would actually be a performance drop to switch it out

Reply:
@NickAragua 
Thoughts on the above?

Reply:
I suppose it's unlikely that you'll encounter too many name overlaps like this. May as well leave it.